### ISSUE TITLE
firefox audiostreams persist even though no audio is playing off firefox

### ISSUE THREAD

**Issue by MrIvanPlays** [wwmm/easyeffects#2024]
### EasyEffects Version

7.0.0

### What package are you using?

Arch (easyeffects)

### Distribution

Arch Linux

### Describe the bug

Basically the title.
Because of this, firefox keeps freezing at random. Closing easyeffects fixes the issue. I expect this is the reason why it happens.

Here's an image show off (no firefox tabs that play audio are currently opened):
![image](https://img.mrivanplays.com/vhsjcpirha.png)

### Expected Behavior

It should not keep them if the tab from where the audio source is coming from is closed/paused.

**Comment by wwmm** [comment]
>It should not keep them if the tab from where the audio source is coming from is closed/paused.

No. This is not how things work. The stream is there because Firefox is not finishing them. EasyEffects is not the one creating these streams. Even if we wanted we can not kill them. This is between Firefox and PipeWire.

>Closing easyeffects fixes the issue. I expect this is the reason why it happens.

Probably a bug in PipeWire. The only difference when EasyEffects is running is that all these streams will be linked to our virtual device instead of the sound card node. But our virtual device is created through a PipeWire `null-sink`. In other words it is PipeWire code.

**Comment by MrIvanPlays** [comment]
> No. This is not how things work. The stream is there because Firefox is not finishing them. EasyEffects is not the one creating these streams. Even if we wanted we can not kill them. This is between Firefox and PipeWire.

So should I report to PipeWire or not?

> > firefox keeps freezing at random
> 
> Is there any related messages in the output of `sudo journalctl -b` when it freezes? Just making Firefox playing to a null-sink should not have this side effect. I use it everyday and I have never seen this happening.

Around 15 minutes before your responses it froze again for no reason at all, but this time it was not EasyEffects. My brain didn't jump into "seeing the logs" mode so I don't know when will it freeze again, nor will I be able to capture any logs.

**Comment by wwmm** [comment]
>So should I report to PipeWire or not?

Yes. Specially if it is also happening without EasyEffects. I think that the combination of null-sink and filters is only making it easier to see a problem that is located somewhere else.

>My brain didn't jump into "seeing the logs" mode so I don't know when will it freeze again, nor will I be able to capture any logs.

Depending on how your system is configured and if logs were printed by Firefox or PipeWire you should still have them in the output of `sudo journalctl`. The difference is that in this case you may see the logs of several days.

**Comment by wwmm** [comment]
So much has changed that is probably better to close this issue
